bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/AccessibleToolBar.java

```diff
@@ -49,34 +51,25 @@ public AccessibleToolBar(Composite parent) {
 	 * @param styleBits the StyleBits to apply to the created ToolItem
 	 * @return a newly created ToolItem
 	 */
-	public ToolItem createToolItem(int styleBits) {
-		ToolBar parent = new ToolBar(this, SWT.FLAT | SWT.HORIZONTAL);
-		ToolItem toolItem = new ToolItem(parent, styleBits);
-
-		addToolItemTraverseListener(parent, toolItem);
-
-		((GridLayout) getLayout()).numColumns++;
-
-		toolBars.add(parent);
+	public AccessibleToolItem createToolItem(int styleBits) {
+		AccessibleToolItem toolItem = new AccessibleToolItem(this, styleBits);
+		accessibleToolItems.add(toolItem);
+		layout.numColumns++;
 		return toolItem;
 	}
 
-	private void addToolItemTraverseListener(ToolBar parent, ToolItem result) {
-		parent.addTraverseListener(e -> {
-			if (e.keyCode == SWT.CR || e.keyCode == SWT.KEYPAD_CR) {
-				result.setSelection(!result.getSelection());
-				e.doit = false;
-			}
-		});
-	}
-
 	@Override
 	public void setBackground(Color color) {
 		super.setBackground(color);
-		for (ToolBar bar : toolBars) { // some ToolItems (like SWT.SEPARATOR) don't easily inherit the color from the
-										// parent control.
-			bar.setBackground(color);
+		// some ToolItems (like SWT.SEPARATOR) don't easily inherit the color from the
+		// parent control
+		for (AccessibleToolItem item : accessibleToolItems) {
+			item.setBackground(color);
```

Comment:
```suggestion
		accessibleToolItems.forEach(item -> item.setBackground(color));
```

in analogy to https://github.com/eclipse-platform/eclipse.platform.ui/pull/2078/files#diff-039028c8c1559043f83229f2942af583d01c16a791712455f9a3ca7abf596997R71

Reply:
Thanks! I actually changed the other code instead of this one. I am not a fan of the `Iterable.forEach()` as it unnecessarily complicates stack traces and debugging. But that's kind of personal preference.